File: orders-protocol/src/main/scala/bookingtour/protocols/orders/api/PointOption.scala
package bookingtour.protocols.orders.api

import bookingtour.protocols.core._
import bookingtour.protocols.core.values.enumeration.PointItem
import bookingtour.protocols.parties.newTypes.PointId
import cats.instances.all._
import derevo.cats.order
import derevo.circe.{decoder, encoder}
import derevo.derive

/**
  * © Alexey Toroshchin 2020.
  */
@derive(encoder, decoder, order)
final case class PointOption(value: PointId, category: PointItem, label: String)

File: customer-api-spa/types/geo/PointOption.ts
import { PointItem } from './PointItem'
import { PointUI } from './PointUI'

export type PointOption = {
  value: string;
  category: PointItem;
  label: string;
}

export function toPointOption(x: PointUI): PointOption {
  return {
    value: x.id,
    category: x.category,
    label: x.label.label
  }
}